include/boost/geometry/algorithms/for_each.hpp

```diff
@@ -109,6 +109,9 @@ struct fe_range_per_segment_with_closure
         typedef typename boost::range_iterator<Range>::type iterator_type;
 
         iterator_type it = boost::begin(range);
+        if (it == boost::end(range))
+            return;
```

Comment:
Thank you Mateusz! Can you add curly braces around `return;`?

Reply:
Sure, done.

Reply:
Thanks